app/controllers/articles_controller.rb

```diff
@@ -27,23 +31,26 @@ def edit
 
   def update
     @article = Article.find(params[:id])
-
     if @article.update(article_params)
       redirect_to @article
     else
       render :edit, status: :unprocessable_entity
     end
   end
 
+
   def destroy
     @article = Article.find(params[:id])
-    @article.destroy
-
-    redirect_to root_path, status: :see_other
+    if @article.discarded? 
```

Comment:
Curious what the purpose of this method is? Why does it un-discard if it should be discarding?